### PR TITLE
Pass the current BinlogPosition to DMLEvent so BinlogWriter can set lastWrittenBinlogPosition

### PR DESCRIPTION
**Pull request by shuhaowu**
Since the BinlogWriter is a separate goroutine to the BinlogStreamer, the position that it wrote to is important if we want to interrupt and resume Ghostferry (interrupt saves BinlogWriter position, and resume starts the BinlogStreamer from the BinlogWriter position).

To allow for this, we need to pass the binlog position from the binlog streamer, via the DMLEvent, to the binlog writer. This patch adds BinlogPosition() to DMLEvent to make this possible.

### CODE REVIEW

File: binlog_writer.go [hunk 1]
```diff
@@ -109,5 +109,8 @@ func (b *BinlogWriter) writeEvents(events []DMLEvent) error {
 	if err != nil {
 		return fmt.Errorf("exec query (%d bytes): %v", len(query), err)
 	}
+
+	// lastWrittenBinlogPosition = events[len(events)-1].BinlogPosition()
```
Comment (shuhaowu) [inline binlog_writer.go]: This is temporary. Once this merges, I'll change this line in #60 such that we send this position to the StateTracker.

File: test/binlog_streamer_test.go [hunk 2]
```diff
@@ -70,6 +83,35 @@ func (this *BinlogStreamerTestSuite) TestConnectErrorsOutIfErrorInServerIdGenera
...
+func (this *BinlogStreamerTestSuite) TestBinlogStreamerSetsBinlogPositionOnDMLEvent() {
+	err := this.binlogStreamer.ConnectBinlogStreamerToMysql()
+	this.Require().Nil(err)
+
+	eventAsserted := false
+
+	this.binlogStreamer.AddEventListener(func(evs []ghostferry.DMLEvent) error {
+		eventAsserted = true
+		this.Require().Equal(1, len(evs))
+		this.Require().NotEqual("", evs[0].BinlogPosition().Name)
```
Comment (hkdsun) [inline test/binlog_streamer_test.go]: If you wanna make this more solid `SELECT @@log_bin_basename` should give you the basename that you can then `assertMatch`

File: test/binlog_streamer_test.go [hunk 3]
```diff
@@ -70,6 +83,35 @@ func (this *BinlogStreamerTestSuite) TestConnectErrorsOutIfErrorInServerIdGenera
...
+		this.Require().NotEqual(0, evs[0].BinlogPosition().Pos)
```
Comment (hkdsun) [inline test/binlog_streamer_test.go]: can you assert(pos > 0) instead?

**Comment by hkdsun** [comment]
thanks for making this easier to review ❤️

it's a huge help when we're not primarily focused on this!

File: test/binlog_streamer_test.go (hunk 2 above)
Comment (shuhaowu) [reply test/binlog_streamer_test.go]: We actually set `--log-bin=mysql-bin` in the docker-compose config so I'll just match against mysql-bin directly. Thanks for the suggestion.

**Pull request merged**